### PR TITLE
add new command couch-config get/set

### PR DESCRIPTION
**Pull request by garrensmith**
Get - will get all the configs settings for all nodes in a cluster and
display them for easy viewing

Set - will set the config for all nodes in a cluster

### CODE REVIEW

File: src/couch-config.js [hunk 1]
```diff
@@ -0,0 +1,160 @@
...
+    err.type = 'EUSAGE';
+    throw err;
+  }
+
+  return nodes;
+}
+
+export function get (cluster, nodes, section) {
+  const promise = Promise.reduce(Object.keys(nodes), (obj, node) => {
+    let url = buildConfigUrl(node, nodes[node], section);
```
Comment (robertkowalski) [inline src/couch-config.js]: this can be a const

File: src/couch-config.js [hunk 2]
```diff
@@ -0,0 +1,160 @@
...
+      if (!results[node]) {
+        return true;
+      }
+
+      return false
+    });
+
+    if (offline.length > 0) {
+      const msg = offline.map(node => 'Node ' + offline + ' is offline.').join('');
+      var err = new Error(msg);
```
Comment (robertkowalski) [inline src/couch-config.js]: this can be a const

File: src/couch-config.js [hunk 3]
```diff
@@ -0,0 +1,160 @@
...
+    });
+
+    const allPromise = Promise.all(promises);
+    allPromise
+    .then((resp) => {
+      console.log(prettyjson.render(resp));
+    })
+    .catch((err) => {
+      console.log('ERR', err);
+      throw err;
```
Comment (robertkowalski) [inline src/couch-config.js]: i think just throwing is enough, the rest will get logged by npmlog when the excpetion is caught

File: src/couch-config.js [hunk 4]
```diff
@@ -0,0 +1,160 @@
...
+    if (er) {
+      er.type = 'EUSAGE';
+      return reject(er);
+    }
+    const cleanedUrl = utils.removeUsernamePw(url);
+    log.http('request', 'PUT', cleanedUrl);
+
+    Wreck.put(url, {payload: JSON.stringify(value)}, (err, res, payload) => {
+      if (err) {
+        let error = new Error('Error on set config for node ' + node + ' ' + err);
```
Comment (robertkowalski) [inline src/couch-config.js]: const!

File: src/couch-config.js [hunk 5]
```diff
@@ -0,0 +1,160 @@
...
+
+  return configUrl;
+}
+
+export function getConfig (node, url) {
+  return new Promise((resolve, reject) => {
+    let er = utils.checkUrl(url);
+
+    if (!er && !/^(http:|https:)/.test(url)) {
+      er = new Error('invalid protocol, must be https or http');
```
Comment (robertkowalski) [inline src/couch-config.js]: what do you think of extending `utils.checkUrl` ?

There library used for it has support for it: 

https://github.com/apache/couchdb-nmo/blob/master/node_modules/valid-url/index.js#L10-L11

Comment (robertkowalski) [inline src/couch-config.js]: it is also used in isonline.js this way, maybe it's a separate PR

File: src/couch-config.js [hunk 6]
```diff
@@ -0,0 +1,160 @@
...
+    });
+
+    if (offline.length > 0) {
+      const msg = offline.map(node => 'Node ' + offline + ' is offline.').join('');
+      var err = new Error(msg);
+      err.type = 'EUSAGE';
+      throw err;
+    }
+
+    var promises = Object.keys(nodes).map(node => {
```
Comment (robertkowalski) [inline src/couch-config.js]: use const instead

File: src/couch-config.js (hunk 3 above)
Comment (robertkowalski) [inline src/couch-config.js]: background info: `console.log` logs on stdout which is not intended for error output

File: src/couch-config.js (hunk 5 above)
Comment (garrensmith) [inline src/couch-config.js]: Its a good idea. But a separate PR. I can do that tomorrow.

Comment (garrensmith) [inline src/couch-config.js]: I decided to refactor it in this PR. It was a quick fix in the end.

File: src/couch-config.js [hunk 7]
```diff
@@ -0,0 +1,151 @@
...
+import Promise from 'bluebird';
+import prettyjson from 'prettyjson';
+import nmo from './nmo.js';
+import isonline, { getClusterUrls } from './isonline.js';
+
+export function cli (cmd, cluster, section, key, value) {
+  return new Promise((resolve, reject) => {
+
+    if (!cmd || !cluster || !exports[cmd]) {
+      const err = new Error('Usage: nmo couch-config add/set <cluster> ...');
```
Comment (robertkowalski) [inline src/couch-config.js]: one small thing, can you change the usage message that it explains both commands, like in https://github.com/apache/couchdb-nmo/blob/35982a557a61ef04bb2ff66302dfe29be2e78fea/src/cluster.js#L24

**Comment by robertkowalski** [comment]
+1 after changing the message

**Comment by garrensmith** [comment]
Merged.

**Pull request closed without merging**